src/context/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { createContext, ReactNode, useContext, FC, useState } from 'react'
+import { useRouter } from 'next/router';
+
+interface Props {
+    children?: ReactNode;
+}
+
+export type ContextType = {
+    state: { toggleEditButton: boolean };
+    actions: { handleToggleButton: () => void }
+}
+const AppContext = createContext<ContextType | null>(null)
+
+const AppWrapper: FC<Props> = ({ children }) => {
+    const router = useRouter()
+    const [toggleEditButton, setToggleEditButton] = useState(false);
+    const handleToggleButton = () => {
+        setToggleEditButton(prev => {
+            const updatedState = !prev;
+            if (updatedState) {
+                router.push('/?edit=true')
+            } else {
+                router.push('/')
+            }
+            return updatedState
+        })
+    }
+    return (
+        <AppContext.Provider value={{
+            state: {
+                toggleEditButton
+            },
+            actions: {
+                handleToggleButton
+            }
+        }}>
+            {children}
+        </AppContext.Provider>
+    )
+}
+
+export const useAppContext = () => {
+    const context = useContext(AppContext) as ContextType
```

Comment:
What if we have not defined any context ?
Shouldn't we be handling that issue?
    if(!context)
        throw new Error(CUSTOM_ERROR_MESSAGE)
    return context

Reply:
Define an error message as AppContext in this case is not defined to ensure that we track the error message and reach to this context